mlrun/datastore/utils.py

```diff
@@ -91,8 +91,12 @@ def filter_df_start_end_time(
     start_time: pd.Timestamp = None,
     end_time: pd.Timestamp = None,
 ) -> typing.Union[pd.DataFrame, typing.Iterator[pd.DataFrame]]:
-    if not time_column or (not start_time and not end_time):
+    if not time_column:
         return df
+    if not start_time and not end_time:
+        df[time_column] = pd.to_datetime(df[time_column])
```

Comment:
can you explain more why you need to do this?
do we need to this always? ie without respect to the type of df[time_column]?

Reply:
This code is executed in `_execute_time_filter`. Actually, it's not necessary here, and not correct in all cases. The right solution is to remove `or (not start_time and not end_time)` from the condition, so `_execute_time_filter` will still run.

Either way, the reason is that we need to cast the time column to a datetime type, as it may be a string.